rotkehlchen/db/dbhandler.py

```diff
@@ -706,18 +708,27 @@ def get_dynamic_cache(
     ) -> Timestamp | None:
         ...
 
+    @overload
+    def get_dynamic_cache(
+            self,
+            cursor: 'DBCursor',
+            name: Literal[DBCacheDynamic.EXTRA_INTERNAL_TX],
+            **kwargs: Unpack[ExtraTxArgType],
+    ) -> ChecksumEvmAddress | None:
+        ...
+
     def get_dynamic_cache(
             self,
             cursor: 'DBCursor',
             name: DBCacheDynamic,
             **kwargs: str,
-    ) -> Timestamp | None:
+    ) -> Timestamp | ChecksumEvmAddress | None:
         """Returns the cache value from the `key_value_cache` table of the DB
         according to the given `name` and `kwargs`. Defaults to `None` if not found."""
         value = cursor.execute(
             'SELECT value FROM key_value_cache WHERE name=?;', (name.get_db_key(**kwargs),),
         ).fetchone()
-        return None if value is None else Timestamp(int(value[0]))
+        return None if value is None else name.get_typed_value(value[0])
```

Comment:
no this is not a good solution. You are adding an extra if, and basically move the deserialization inside the getter function. 

While the caller of `get_dynamic_cache` knows what to expect and should just use it properly. Can't you just return value[0] and then the caller deserializes properly?

Also again I ask is the timestamp saved as an int or as a string in the DB cache?